File: java/integrations/okhttp/src/main/java/io/zrz/httpsig/okhttp/OkHttpMessage.java
package io.zrz.httpsig.okhttp;

import java.net.URI;
import java.util.List;

import io.zrz.httpsig.HttpMessage;

import okhttp3.Request;

/**
 * Adapts an OkHttp {@link Request} to {@link HttpMessage} for signing.
 */
final class OkHttpMessage implements HttpMessage {

    private final Request request;

    OkHttpMessage(Request request) {
        this.request = request;
    }

    @Override
    public boolean isRequest() {
        return true;
    }

    @Override
    public String method() {
        return request.method();
    }

    @Override
    public URI url() {
        return request.url().uri();
    }

    @Override
    public int statusCode() {
        throw new UnsupportedOperationException("not a response");
    }

    @Override
    public List<String> headerValues(String name) {
        return request.headers(name);
    }
}

File: kotlin/integrations/okhttp/src/main/kotlin/io/zrz/httpsig/okhttp/OkHttpMessage.kt
package io.zrz.httpsig.okhttp

import io.zrz.httpsig.HttpMessage
import okhttp3.Request
import java.net.URI

/**
 * Adapts an OkHttp [Request] to the [HttpMessage] interface for signature operations.
 */
internal class OkHttpMessage(private val request: Request) : HttpMessage {

    override val isRequest: Boolean = true

    override val method: String = request.method

    override val url: URI = request.url.toUri()

    override val statusCode: Int = 0

    override fun headerValues(name: String): List<String> =
        request.headers(name)
}
